### DISCUSSION TITLE
[Question] How to avoid congested/stuck roads?

### DISCUSSION THREAD

**Discussion by juj** [nicymike/serflings#79] [Q&A] [answered]
When playing online with a friend, we observed congested roads to HQ in late game - possibly even the roads getting stuck.

Tried to play a singleplayer game where I'd build a big road network right outside the HQ building to give plenty of room for goods to flow, but it looks like it will result in the congestion moving one or two road nodes farther:

![image](https://user-images.githubusercontent.com/225351/179729997-e9d53721-c76b-4243-ac30-09207eeea409.png)

In this image, the serfs are not moving the goods, but are stuck standing idle, and the intersections pile up stuff.

Tried building warehouses, but not sure how to use them.

Not necessarily a huge deal, the game is still fun, though I'm curious if there are ways to plan to avoid congestion, or if it is inevitable?

**Comment by nicymike** [comment, marked as answer]
Well, in my opinion transportation and building paths (roads) is sometimes a bit challenging. At first glance I think there are too many paths around the castle.

Transportation is working like this:
Calculate the shortest path by number of of segments (from flag to flag) and ignore directions with no carriers or with more than seven goods waiting. The actual length does not count (important to know). Even the game manual does describe this as a way to control the transportation by placing more or less flags.

Also remember that different paths will only be used if a path is stuck so no need to build plenty of paths.

Building stocks is good for reducing the number of goods on paths. If there are too many goods laying around, it may be good to build a stock near to prevent the transportation of goods over long distances.

**Comment by juj** [comment]
Thanks for the tips. My idea there was to immediately build a spanning star pattern out of the starting square, and then a second high fan out spanning star pattern from all of the flags that connect out from there. That way I thought there would be a more free flowing network that might avoid the critical HQ square.

But in the end that seemed to just shift the critical bottleneck from right in front of the HQ to one square to its right.

I'll need to give the manual a good read. At a quick glance I now see it has tons of text beyond just on the controls on how to play the game.

**Comment by nicymike** [comment]
Settlers 1 is one of the games where you need to read the manual to know many of the internal behaviour of the game (eg. win conditions). But I have more ideas for Serflings to help playing the game without the need to read the manual.
